src/contracts/Router.sol

```diff
@@ -0,0 +1,122 @@
+// The Licensed Work is (c) 2022 Sygma
+// SPDX-License-Identifier: LGPL-3.0-only
+pragma solidity 0.8.11;
+
+import "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
+import "@openzeppelin/contracts/utils/Context.sol";
+
+
+import "./interfaces/IAccessControlSegregator.sol";
+import "./interfaces/IBridge.sol";
+import "./interfaces/IHandler.sol";
+import "./interfaces/IFeeHandler.sol";
+
+
+
+/**
+    @title Facilitates proposal deposits.
+    @author ChainSafe Systems.
+ */
+contract Router is Context {
+    using ECDSA for bytes32;
+
+    IBridge public immutable _bridge;
+    IAccessControlSegregator public _accessControl;
```

Comment:
Should there be multiple access controls? Separate for Bridge and Router? Or maybe a single one on Bridge is enough?